frontend/components/ButtonField/ButtonField.js

```diff
@@ -45,7 +45,8 @@ class ButtonField extends React.Component {
                     title="Send own data"
                     disabled={
                         this.props.connectionType === 'wifi' &&
-                        !(process.env.NODE_ENV === 'development')
+                        !(process.env.NODE_ENV === 'development') ||
+                        (this.props.provider === 'Undefinied')
```

Comment:
Furthermore the boolean expression seems to be incorrect. 
The following scenario would lead to an falsy solution: 
* we are in production (true)
* we have no wifi connection (true)
* we have no provider choosen (false)

this would lead to `true && true || false ==> true` so the button would be enabled.